### PR TITLE
Infra - Pi Cross-Compiling with Docker

### PR DESCRIPTION
**Pull request by DylanCavers**
Add functionality to cross-compile for raspberry pi's to the Docker build system.

### CODE REVIEW

File: entry.sh [hunk 1]
```diff
@@ -16,7 +14,15 @@ elif [ ! -d "build" ]; then
...
 
-cd build 
-cmake ..
-make -j
-make test
+if [[ $CROSS_COMPILE = true ]]; then
+    echo "Cross compiling for Raspberry Pi..."
+    cd build 
+    cmake  -DCMAKE_CXX_COMPILER=aarch64-rpi3-linux-gnu-g++ -DCURSES_LIBRARY=/usr/lib/x86_64-linux-gnu/libncurses.so -DCURSES_INCLUDE_PATH=/usr/include ..
+    make -j
```
Comment (TomLonergan03) [inline entry.sh:21]: cos this is an arm container, you can run `make test` also right?

Comment (DylanCavers) [reply entry.sh:21]: pretty sure we can, but may be overkill as we presumably won't need to run the tests at this stage and will be fairly computationally expensive cause of emulation. Unless we want it as proof the cross-compiled code works.

File: docker.sh [hunk 2]
```diff
@@ -83,22 +76,62 @@ handle_options() {
...
+    docker buildx build -f cc/Dockerfile.crosscompile -t $CC_IMAGE_NAME .
+  else
+    echo "[!] Rebuild image"
+    docker build -t $IMAGE_NAME .
+  fi
+else
+  # Only build if the image does not exist
+  if [ "$cross_compile" = true ]; then
+    image=$( docker images -q $CC_IMAGE_NAME 2> /dev/null )
+    echo $image
```
Comment (davidbeechey) [inline docker.sh:93]: was this just for debugging?

File: cc/Dockerfile.crosscompile
Comment (davidbeechey) [inline cc/Dockerfile.crosscompile:1]: is there a reason this is in its own folder?

**Review by davidbeechey — APPROVED** [review]
Looks good :)

File: docker.sh (hunk 2 above)
Comment (DylanCavers) [reply docker.sh:93]: ah yeah will remove

File: cc/Dockerfile.crosscompile
Comment (DylanCavers) [reply cc/Dockerfile.crosscompile:1]: initially for neatness as I had config files etc. But now its kinda pointless, can either clean up the other docker stuff into a Docker folder or move this dockerfile out to the root directory

File: entry.sh [hunk 3]
```diff
@@ -18,5 +18,5 @@ fi
 
 cd build 
 cmake ..
-make
+make -j4
```
Comment (TomLonergan03) [inline entry.sh:21]: just `-j` or do some experimenting to find the fastest number based on `nprocs` (I've heard `nprocs`/`nprocs+1`/`nprocs*2` are all potentials)

**Review by TomLonergan03 — CHANGES_REQUESTED** [review]

File: entry.sh (hunk 3 above)
Comment (DylanCavers) [reply entry.sh:21]: cool will investigate, `-j4` was the highest I could go without it failing

**Review by TomLonergan03 — APPROVED** [review]
have u tested with a pi?

**Pull request merged**